From e7439f5fff042d3a498197f5793dbef331d788cf Mon Sep 17 00:00:00 2001
From: Anish Trivedi <anish@freescale.com>
Date: Thu, 31 Mar 2011 15:57:16 -0500
Subject: [PATCH 140/280] ENGR00141479 [imx-lib] SAHARA: Use relocated header
 file

The user mode libsahara library relied on header file
drivers/mxc/security/sahara2/include/sahara.h. However, to make
the lib build after headers_install step and to remove dependency on
kernel source, this header was moved to include/linux/mxc_sahara.h.

Made necessary changes to reference the relocated header file in
imx-lib.

Signed-off-by: Anish Trivedi <anish@freescale.com>
---
 sahara2/fsl_shw_auth.c | 4 ++--
 sahara2/fsl_shw_hash.c | 4 ++--
 sahara2/fsl_shw_hmac.c | 4 ++--
 sahara2/fsl_shw_rand.c | 4 ++--
 sahara2/fsl_shw_sym.c  | 4 ++--
 sahara2/fsl_shw_user.c | 4 ++--
 sahara2/fsl_shw_wrap.c | 4 ++--
 sahara2/um_adaptor.c   | 5 ++---
 8 files changed, 16 insertions(+), 17 deletions(-)

diff --git a/sahara2/fsl_shw_auth.c b/sahara2/fsl_shw_auth.c
index 185ef77..5e2d033 100644
--- a/sahara2/fsl_shw_auth.c
+++ b/sahara2/fsl_shw_auth.c
@@ -1,6 +1,6 @@
 /*
  * User Space library to access the Security hardware
- * Copyright 2005-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2005-2009, 2011 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This library is free software; you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -25,7 +25,7 @@
  * functions.  For now, only AES-CCM is supported.
  */
 
-#include "sahara.h"
+#include <linux/mxc_sahara.h>
 #include "adaptor.h"
 #include "sf_util.h"
 
diff --git a/sahara2/fsl_shw_hash.c b/sahara2/fsl_shw_hash.c
index db33811..abac654 100644
--- a/sahara2/fsl_shw_hash.c
+++ b/sahara2/fsl_shw_hash.c
@@ -1,6 +1,6 @@
 /*
  * User Space library to access the Security hardware
- * Copyright 2005-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2005-2009, 2011 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This library is free software; you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -25,7 +25,7 @@
  * for Sahara.  This does not include HMAC.
  */
 
-#include "sahara.h"
+#include <linux/mxc_sahara.h>
 #include "sf_util.h"
 
 #ifdef LINUX_KERNEL
diff --git a/sahara2/fsl_shw_hmac.c b/sahara2/fsl_shw_hmac.c
index c513b26..5d3f440 100644
--- a/sahara2/fsl_shw_hmac.c
+++ b/sahara2/fsl_shw_hmac.c
@@ -1,6 +1,6 @@
 /*
  * User Space library to access the Security hardware
- * Copyright (C) 2005-2006 written by Freescale Semiconductor
+ * Copyright (C) 2005-2006, 2011 Freescale Semiconductor, Inc.
  *
  * This library is free software; you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -25,7 +25,7 @@
  * SHW API for Sahara.
  */
 
-#include "sahara.h"
+#include <linux/mxc_sahara.h>
 #include "sf_util.h"
 
 #ifdef __KERNEL__
diff --git a/sahara2/fsl_shw_rand.c b/sahara2/fsl_shw_rand.c
index 423f3c5..301df0a 100644
--- a/sahara2/fsl_shw_rand.c
+++ b/sahara2/fsl_shw_rand.c
@@ -1,6 +1,6 @@
 /*
  * User Space library to access the Security hardware
- * Copyright 2005-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2005-2009, 2011 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This library is free software; you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -25,7 +25,7 @@
  * for Sahara.
  */
 
-#include "sahara.h"
+#include <linux/mxc_sahara.h>
 #include "sf_util.h"
 
 #ifdef __KERNEL__
diff --git a/sahara2/fsl_shw_sym.c b/sahara2/fsl_shw_sym.c
index 4d7f410..56455c5 100644
--- a/sahara2/fsl_shw_sym.c
+++ b/sahara2/fsl_shw_sym.c
@@ -1,6 +1,6 @@
 /*
  * User Space library to access the Security hardware
- * Copyright 2005-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2005-2009, 2011 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This library is free software; you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -25,7 +25,7 @@
  * Sahara.  This does not include CCM.
  */
 
-#include "sahara.h"
+#include <linux/mxc_sahara.h>
 #include "fsl_platform.h"
 #include "sf_util.h"
 #include "adaptor.h"
diff --git a/sahara2/fsl_shw_user.c b/sahara2/fsl_shw_user.c
index a2ded23..26c36bd 100644
--- a/sahara2/fsl_shw_user.c
+++ b/sahara2/fsl_shw_user.c
@@ -1,6 +1,6 @@
 /*
  * User Space library to access the Security hardware
- * Copyright 2005-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2005-2009, 2011 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This library is free software; you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -24,7 +24,7 @@
  * This file implements user and platform capabilities functions of the FSL SHW
  * API for Sahara
  */
-#include "sahara.h"
+#include <linux/mxc_sahara.h>
 #include <adaptor.h>
 #include <sf_util.h>
 
diff --git a/sahara2/fsl_shw_wrap.c b/sahara2/fsl_shw_wrap.c
index d5ebf90..fa2b178 100644
--- a/sahara2/fsl_shw_wrap.c
+++ b/sahara2/fsl_shw_wrap.c
@@ -1,6 +1,6 @@
 /*
  * User Space library to access the Security hardware
- * Copyright 2005-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2005-2009, 2011 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This library is free software; you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -71,7 +71,7 @@
 
  */
 
-#include "sahara.h"
+#include <linux/mxc_sahara.h>
 #include "fsl_platform.h"
 #include "fsl_shw_keystore.h"
 
diff --git a/sahara2/um_adaptor.c b/sahara2/um_adaptor.c
index c39deeb..8ef2ec9 100644
--- a/sahara2/um_adaptor.c
+++ b/sahara2/um_adaptor.c
@@ -1,6 +1,6 @@
 /*
  * User Space library to access the Security hardware
- * Copyright 2008-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2008-2009, 2011 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This library is free software; you can redistribute it and/or modify
  * it under the terms of the GNU Lesser General Public License as
@@ -38,9 +38,8 @@
 #include <pthread.h>
 #include <stdio.h>
 
-#include <sahara.h>
+#include <linux/mxc_sahara.h>
 #include <sah_kernel.h>
-#include <sahara.h>
 #include <adaptor.h>
 
 #include <memory.h>
-- 
1.8.0

